src/UhdmAst.cpp

```diff
@@ -2209,6 +2209,14 @@ namespace UhdmAst {
                   fromp,
                   ((AstSelExtract*)item)->msbp()->cloneTree(true),
                   ((AstSelExtract*)item)->lsbp()->cloneTree(true));
+            } else if(item->type() == AstType::en::atConst) {
+              select = new AstSelBit(new FileLine("uhdm"), fromp, bitp);
+            } else if (item->type() == AstType::atSelPlus) {
```

Comment:
Can we handle `-:` selection here as well? Only the node type would change.

Reply:
Added `SelMinus`